packages/cojson-storage-indexeddb/src/index.ts

```diff
@@ -1,733 +1 @@
-              )
-            : SyncPromise.resolve([])
-        ).then((allOurSessions) => {
-          const ourKnown: CojsonInternalTypes.CoValueKnownState = {
-            id: theirKnown.id,
-            header: !!coValueRow,
-            sessions: {},
-          };
-
-          const newContentPieces: CojsonInternalTypes.NewContentMessage[] = [
-            {
-              action: "content",
-              id: theirKnown.id,
-              header: theirKnown.header ? undefined : coValueRow?.header,
-              new: {},
-              priority: cojsonInternals.getPriorityFromHeader(
-                coValueRow?.header,
-              ),
-            },
-          ];
-
-          return SyncPromise.all(
-            allOurSessions.map((sessionRow) => {
-              ourKnown.sessions[sessionRow.sessionID] = sessionRow.lastIdx;
-
-              if (
-                sessionRow.lastIdx >
-                (theirKnown.sessions[sessionRow.sessionID] || 0)
-              ) {
-                const firstNewTxIdx =
-                  theirKnown.sessions[sessionRow.sessionID] || 0;
-
-                return this.makeRequest<SignatureAfterRow[]>(
-                  ({ signatureAfter }) =>
-                    signatureAfter.getAll(
-                      IDBKeyRange.bound(
-                        [sessionRow.rowID, firstNewTxIdx],
-                        [sessionRow.rowID, Infinity],
-                      ),
-                    ),
-                ).then((signaturesAndIdxs) => {
-                  // console.log(
-                  //     theirKnown.id,
-                  //     "signaturesAndIdxs",
-                  //     JSON.stringify(signaturesAndIdxs)
-                  // );
-
-                  return this.makeRequest<TransactionRow[]>(
-                    ({ transactions }) =>
-                      transactions.getAll(
-                        IDBKeyRange.bound(
-                          [sessionRow.rowID, firstNewTxIdx],
-                          [sessionRow.rowID, Infinity],
-                        ),
-                      ),
-                  ).then((newTxsInSession) => {
-                    collectNewTxs(
-                      newTxsInSession,
-                      newContentPieces,
-                      sessionRow,
-                      signaturesAndIdxs,
-                      theirKnown,
-                      firstNewTxIdx,
-                    );
-                  });
-                });
-              } else {
-                return SyncPromise.resolve();
-              }
-            }),
-          ).then(() => {
-            const dependedOnCoValues = getDependedOnCoValues(
-              coValueRow,
-              newContentPieces,
-              theirKnown,
-            );
-
-            return SyncPromise.all(
-              dependedOnCoValues.map((dependedOnCoValue) =>
-                this.sendNewContentAfter(
-                  {
```

Comment:
Here we could send a coValue as many times as the recursion repeats even if it was sent earlier with the earliest iterations